### PR TITLE
[WIP] Libupnp fix crash

### PR DESCRIPTION
**Pull request by G10h4ck**
Few a few crashes reported at https://github.com/RetroShare/RetroShare/issues/1837

### CODE REVIEW

File: libretroshare/src/rs_upnp/UPnPBase.cpp [hunk 1]
```diff
@@ -1372,10 +1335,11 @@ int CUPnPControlPoint::Callback(
...
 	{
 #if (UPNP_VERSION > 10624) && (UPNP_VERSION < 10800)
-		std::cerr <<  "********************************************************************************" << std::endl
-		          <<  "***                THIS SHOULD NOT HAPPEN !!!  TELL IT TO DEVS               ***" << std::endl
-		          <<  "*** UPnPBase.cpp CUPnPControlPoint::Callback() UPNP_DISCOVERY_SEARCH_TIMEOUT ***" << std::endl
-		          <<  "********************************************************************************" << std::endl;
+		RsErr() << __PRETTY_FUNCTION__
+		        << " ******** (UPNP_VERSION > 10624) && (UPNP_VERSION < 10800) "
+		           " UPNP_DISCOVERY_SEARCH_TIMEOUT  ***************************"
+		           " THIS SHOULD NEVER HAPPEN !!!  TELL IT TO DEVS ************"
```
Comment (G10h4ck) [inline libretroshare/src/rs_upnp/UPnPBase.cpp]: @PhenomRetroShare  This needs clarification https://github.com/RetroShare/RetroShare/issues/1996

Comment (PhenomRetroShare) [reply libretroshare/src/rs_upnp/UPnPBase.cpp]: Look at this comment:
https://github.com/RetroShare/RetroShare/pull/1603#issuecomment-515194637
But I don't remember...

**Comment by csoler** [comment]
For the record, it seems that 95% of this PR is about introducing a new debug output. When fixing bugs, it would be really nice if the PRs where only dedicated to fixing that particular bug, because reviewing all these changes makes it much more risky to overlook a mistake.

**Comment by G10h4ck** [comment]
> For the record, it seems that 95% of this PR is about introducing a new debug output. When fixing bugs, it would be really nice if the PRs where only dedicated to fixing that particular bug, because reviewing all these changes makes it much more risky to overlook a mistake.

This is still a WIP (Work In Progress)... Digging into the code I discovered that our way to handle UPnP (at least when using libupnp) is completely flawed...
To fix the bug became a chain reaction :-| and I am quite demotivated about it... I am tempted to build RS with miniupnpc which is not affected, and let it go...

**Comment by PhenomRetroShare** [comment]
LibUPnP needs more reworks since 1.14.0 release.
Read the change log:
https://github.com/pupnp/pupnp/blob/master/ChangeLog#L43

**Comment by sehraf** [comment]
> LibUPnP needs more reworks since 1.14.0 release.

related to #2072

**Comment by G10h4ck** [comment]
Yeah libupnp have been changing it's API a lot lately, I am a bit pissed off of keeping adding `#ifdef` to libretroshare libupnp related code also because this part of our code is quite a mess and have also a few crashes and race conditions because it is badly designed. Even if this time it is easy to make RS compile against latest libupnp how much it will last,and it will keep crashing anyway, I think we should compile against miniupnpc in all platforms it is available until the whole RS libpnp code get overhauled.

**Comment by G10h4ck** [comment]
@sehraf thanks for the snippet in the issue, it looks easy to support last API change for this time, would you mind to test it a bit on Arch which have the latest libupnp? Or your router doesn't have UPnP at all? I might be able to test it in a couple of weeks

**Comment by sehraf** [comment]
As far as I know, my router has no upnp. I have never used it and have never seen any option regarding it.

**Comment by G10h4ck** [comment]
After fiddling with this stuff, my conclusion is that o have a stable UPnP implementation based on libupnp retroshare libupnp interface must be rewritten from scratch as is rely on wrong assumptions, hence i close this PR. For the moment I suggest anyone building RS to use miniupnp or none.

**Pull request closed without merging**